core/schema/internalsdk.graphqls

```diff
@@ -0,0 +1,138 @@
+extend type Query {
+  """
+  Create a new function from the provided definition.
+  """
+  newFunction(def: FunctionDef!): Function!
+
+  """
+  The FunctionCall context that the SDK caller is currently executing in.
+  If the caller is not currently executing in a function, this will return
+  an error.
+  """
+  currentFunctionCall: FunctionCall!
+}
+
+type FunctionCall {
+  "The name of the function being called."
+  name: String!
+
+  """
+  The name of the parent object of the function being called.
+  If the function is "top-level" to the module, this is the name of the module.
+  """
+  parentName: String!
+
+  """
+  The value of the parent object of the function being called.
+  If the function is "top-level" to the module, this is always an empty object.
+  """
+  parent: JSON!
+
+  """
+  The argument values the function is being invoked with.
+  """
+  inputArgs: [FunctionCallArgValue!]
+
+  """
+  Set the return value of the function call to the provided value.
+  The value should be a string of the JSON serialization of the return value.
+  """
+  returnValue(value: JSON!): Void
+}
+
+type FunctionCallArgValue {
+  "The name of the argument."
+  name: String!
+
+  "The value of the argument represented as a string of the JSON serialization."
+  value: JSON!
+}
+
+input FunctionDef {
```

Comment:
The duplication required in order for these to be `input` is slightly sad, but it's part of avoiding the `with*` fields as mentioned in [my other comment](https://github.com/dagger/dagger/pull/5757/files#r1316272095).